### PR TITLE
Ray 1.5.0

### PR DESCRIPTION
**Pull request by krfricke**
After 1.4.1 and before 1.5.1, here's Ray 1.5.0

Build passed locally

cc @vnlitvinov @h-vetinari 

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by krfricke** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -84,31 +85,33 @@ outputs:
...
-        - aiohttp-cors
-        - aioredis
         - click >=7.0
         - colorama
-        - colorful
         - dataclasses  # [py<37]
         - filelock
         # gpustat-0.6.0 has a dependency which does not exist on Windows;
         # skip it there until gpustat is fixed as it is optional
-        - gpustat  # [not win]
```
Comment (h-vetinari) [inline recipe/meta.yaml:97]: The comment should travel with that moved line.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -125,6 +128,20 @@ outputs:
...
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+        - colorful
+
+    test:
+      imports:
+        # there doesn't appear to be a k8s module, not sure how to test this
+        - ray
```
Comment (h-vetinari) [inline recipe/meta.yaml:143]: Please improve this comment.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -215,16 +233,18 @@ outputs:
       run:
         - python
         - {{ pin_subpackage('ray-tune', exact=True) }}
-        - atari_py
         - dm-tree
         - gym
         - lz4
+        - scikit-image
+        - pyyaml
+        - scipy
```
Comment (h-vetinari) [inline recipe/meta.yaml:241]: Where no other overriding considerations exist, please keep packages alphabetized.

**Comment by h-vetinari** [comment]
Thanks a lot for this! :)

BTW, your commit history still refers to 1.5.1, which is confusing. Please rebase or squash this.

File: recipe/meta.yaml (hunk 2 above)
Comment (krfricke) [reply recipe/meta.yaml:143]: Removed (wrongly copied when creating `ray-default`)

File: recipe/meta.yaml [hunk 4]
```diff
@@ -84,31 +85,33 @@ outputs:
...
+        - aiohttp
+        - aiohttp-cors
+        - aioredis <2
+        - jsonschema
+        - requests
+        # gpustat-0.6.0 has a dependency which does not exist on Windows;
+        # skip it there until gpustat is fixed as it is optional
+        - gpustat  # [not win]
+        - opencensus
+        - prometheus_client >=0.7.1
```
Comment (h-vetinari) [inline recipe/meta.yaml:113]: Same thing for alphabetization here

File: recipe/meta.yaml [hunk 5]
```diff
@@ -125,6 +128,19 @@ outputs:
...
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('ray-core', exact=True) }}
+        - colorful
+
+    test:
+      imports:
+        - ray
```
Comment (h-vetinari) [inline recipe/meta.yaml:142]: You should still add a comment when the test import does not match the output name (i.e. `ray-default` here). Something like "ray-default is a convenience wrapper with no imports of its own"

Comment (krfricke) [reply recipe/meta.yaml:143]: Sounds good, added!

**Comment by krfricke** [comment]
I squashed the commit history so it is less confusing

File: recipe/meta.yaml [hunk 6]
```diff
@@ -47,6 +47,7 @@ outputs:
       run:
         - python
         - {{ pin_subpackage('ray-core', exact=True) }}
+        - {{ pin_subpackage('ray-default', exact=True) }}
```
Comment (vnlitvinov) [inline recipe/meta.yaml:50]: shouldn't pinning `ray-default` be enough? I presume that it depends on `ray-core` by itself... and that builds a little easier to follow dependency chain IMHO

**Review by vnlitvinov — APPROVED** [review]
LGTM with one small suggestion

File: recipe/meta.yaml (hunk 6 above)
Comment (h-vetinari) [reply recipe/meta.yaml:50]: Fine by me either way.

**Review by h-vetinari — APPROVED** [review]
Thanks!

**Comment by h-vetinari** [comment]
Let's merge #31 first (Vasilij already added the automerge label); this will create a tiny conflict around the build number, but should be easy to rebase (or we just squash-merge)

**Comment by krfricke** [comment]
Sounds good, I'll address the nit (and resolve the conflict) after #31 has been merged

**Review by h-vetinari — APPROVED** [review]
LGTM; let's squash-merge upon completion of the CI

**Pull request merged**